cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go

```diff
@@ -51,105 +51,160 @@ func (t Template) controllersTrustPolicy() *iamv1.PolicyDocument {
-					"ec2:CreateRoute",
-					"ec2:CreateRouteTable",
-					"ec2:CreateSecurityGroup",
-					"ec2:CreateSubnet",
-					"ec2:CreateTags",
-					"ec2:CreateVpc",
-					"ec2:ModifyVpcAttribute",
-					"ec2:DeleteInternetGateway",
-					"ec2:DeleteNatGateway",
-					"ec2:DeleteRouteTable",
-					"ec2:DeleteSecurityGroup",
-					"ec2:DeleteSubnet",
-					"ec2:DeleteTags",
-					"ec2:DeleteVpc",
-					"ec2:DescribeAccountAttributes",
-					"ec2:DescribeAddresses",
-					"ec2:DescribeAvailabilityZones",
-					"ec2:DescribeInstances",
-					"ec2:DescribeInternetGateways",
-					"ec2:DescribeImages",
-					"ec2:DescribeNatGateways",
-					"ec2:DescribeNetworkInterfaces",
-					"ec2:DescribeNetworkInterfaceAttribute",
-					"ec2:DescribeRouteTables",
-					"ec2:DescribeSecurityGroups",
-					"ec2:DescribeSubnets",
-					"ec2:DescribeVpcs",
-					"ec2:DescribeVpcAttribute",
-					"ec2:DescribeVolumes",
-					"ec2:DetachInternetGateway",
-					"ec2:DisassociateRouteTable",
-					"ec2:DisassociateAddress",
-					"ec2:ModifyInstanceAttribute",
-					"ec2:ModifyNetworkInterfaceAttribute",
-					"ec2:ModifySubnetAttribute",
-					"ec2:ReleaseAddress",
-					"ec2:RevokeSecurityGroupIngress",
-					"ec2:RunInstances",
-					"ec2:TerminateInstances",
-					"tag:GetResources",
-					"elasticloadbalancing:AddTags",
-					"elasticloadbalancing:CreateLoadBalancer",
-					"elasticloadbalancing:ConfigureHealthCheck",
-					"elasticloadbalancing:DeleteLoadBalancer",
-					"elasticloadbalancing:DescribeLoadBalancers",
-					"elasticloadbalancing:DescribeLoadBalancerAttributes",
-					"elasticloadbalancing:DescribeTags",
-					"elasticloadbalancing:ModifyLoadBalancerAttributes",
-					"elasticloadbalancing:RegisterInstancesWithLoadBalancer",
-					"elasticloadbalancing:DeregisterInstancesFromLoadBalancer",
-					"elasticloadbalancing:RemoveTags",
+					"iam:GetRole",
+					"iam:ListAttachedRolePolicies",
+					"iam:DetachRolePolicy",
+					"iam:DeleteRole",
+					"iam:CreateRole",
+					"iam:TagRole",
+					"iam:AttachRolePolicy",
+				},
+				Resource: iamv1.Resources{
+					"arn:aws:iam::*:role/*",
 				},
-			},
-			{
 				Effect: iamv1.EffectAllow,
+			}, {
+				Action: iamv1.Actions{
+					"iam:GetPolicy",
+				},
 				Resource: iamv1.Resources{
-					"arn:*:iam::*:role/aws-service-role/elasticloadbalancing.amazonaws.com/AWSServiceRoleForElasticLoadBalancing",
+					"arn:aws:iam::aws:policy/AmazonEKSClusterPolicy",
 				},
+				Effect: iamv1.EffectAllow,
+			}, {
 				Action: iamv1.Actions{
-					"iam:CreateServiceLinkedRole",
+					"eks:DescribeCluster",
+					"eks:ListClusters",
+					"eks:CreateCluster",
+					"eks:TagResource",
+					"eks:UpdateClusterVersion",
+					"eks:DeleteCluster",
+					"eks:UpdateClusterConfig",
+					"eks:UntagResource",
 				},
-				Condition: iamv1.Conditions{
-					iamv1.StringLike: map[string]string{"iam:AWSServiceName": "elasticloadbalancing.amazonaws.com"},
+				Resource: iamv1.Resources{
+					"arn:aws:eks:*:*:cluster/*",
```

Comment:
Could we restrict this down to the specific account? Although i see elsewhere that * is used. Perhaps a discussion outside the scope of this?

Reply:
IAM can't be scoped beyond an account anyway, so this is fine. Only the trust policies control which foreign IAM roles you can assume, which will need their own roles, etc...